src/bootstrap/compile.rs

```diff
@@ -1161,8 +1161,14 @@ impl Step for Assemble {
             let dst_exe = exe("rust-llvm-dwp", target_compiler.host);
             let llvm_config_bin = builder.ensure(native::Llvm { target: target_compiler.host });
             if !builder.config.dry_run {
-                let llvm_bin_dir = output(Command::new(llvm_config_bin).arg("--bindir"));
-                let llvm_bin_dir = Path::new(llvm_bin_dir.trim());
+                let llvm_bin_dir = if target_compiler.host == build_compiler.host {
+                    let llvm_bin_dir = output(Command::new(llvm_config_bin).arg("--bindir"));
+                    PathBuf::from(llvm_bin_dir.trim())
+                } else {
+                    // if cross-compiling don't copy from the llvm directory for the target instead
+                    builder.llvm_out(target_compiler.host).join("build/bin")
+                };
```

Comment:
So I admit I may just be confused -- but it feels like llvm_bin_dir should ~match with llvm_config_bin returned from native::Llvm invocation? In particular, I would expect `llvm-config` to be in the same directory as `llvm-dwp`. So maybe we should skip the `--bindir` invocation regardless of cross-compilation and just copy `llvm_config_bin.with_filename(src_exe)`?

If that *doesn't* work -- presumably for the reason we started invoking bindir here -- it would be good to figure out why. Because it seems like the new code makes the assumption that it *should* work, which can then presumably fail for whatever reason we started using `--bindir`.

In some sense I guess this gets at "why does `--bindir` not work for cross-compilation"?